### PR TITLE
[tests] Add xtro support for Mac Catalyst. Fixes #10214.

### LINKED ISSUE #10214: Mac Catalyst: make xtro tests work

### PR DESCRIPTION
**Pull request by rolfbjarne**
* Add support to xtro for Mac Catalyst.
* Add all the .todo files so that xtro doesn't fail.
* Bump Objective-Sharpie.

Fixes https://github.com/xamarin/xamarin-macios/issues/10214.

### CODE REVIEW

File: tests/xtro-sharpie/Makefile [hunk 1]
```diff
@@ -50,7 +50,32 @@ XMAC_PCH = macosx$(OSX_SDK_VERSION)-$(XMAC_ARCH).pch
...
+	-exclude ICADevices \
+	-exclude InputMethodKit \
+	-exclude InstallerPlugins \
+	-exclude IOBluetooth \
+	-exclude IOBluetoothUI \
+	-exclude JavaNativeFoundation \
+	-exclude LDAP \
+	-exclude Python \
+	-exclude Quartz \
+	-exclude SecurityInterface \
```
Comment (spouliot) [inline tests/xtro-sharpie/Makefile:75]: are those exclusions to make sharpie work ?
because others do/should exists

**Review by spouliot — CHANGES_REQUESTED** [review]
There's a lot of files that should not be present in the PR, e.g. `MacCatalyst-ruby.todo`

IOW anything we don't support for macOS (or iOS) should not be committed as `.todo` (or even `.ignore`)

Beside the filtering things looks good :)

File: tests/xtro-sharpie/Makefile (hunk 1 above)
Comment (rolfbjarne) [reply tests/xtro-sharpie/Makefile:75]: Yes, this is to make sharpie work (the corresponding headers don't compile with Mac Catalyst).

**Review by spouliot — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Comment by rolfbjarne** [comment]
build

**Comment by rolfbjarne** [comment]
build

**Comment by rolfbjarne** [comment]
build

**Comment by rolfbjarne** [comment]
API comparison failure is unrelated.

**Pull request merged**